silkworm/silkrpc/core/blocks.cpp

```diff
@@ -78,10 +79,13 @@ boost::asio::awaitable<std::pair<uint64_t, bool>> get_block_number(const std::st
     } else if (block_id == kLatestExecutedBlockId) {
         block_number = co_await get_latest_executed_block_number(reader);
         is_latest_block = true;
-    } else {
-        block_number = static_cast<uint64_t>(std::stol(block_id, nullptr, 0));
+    } else if (is_valid_hex(block_id)) {
+        block_number = static_cast<uint64_t>(std::stol(block_id, nullptr, 16));
         check_if_latest = latest_required;
+    } else {
+        throw std::invalid_argument("Invalid Block Id");
     }
+
```

Comment:
Is this change required specifically by some Execution API unit test? If this is not the case, I would prefer the previous conversion because by using `0` as numeric base we allow for both hex and decimal block number in input string.

This version is stricter, which can be fine or maybe better but we need to check all the API implementations using directly or indirectly the `get_block_number` function. Currently, this requires you to run the integration tests, which in turn requires an Erigon core component running on Goerli testnet.

I hope we can move this kind of non-regression validations to our unit test suite for `silkrpc` you're building, but for the time being we need to ask @lupin012 to run the integration tests on this branch.